### ISSUE TITLE
Corrupted data trashes the output

### ISSUE THREAD

**Issue by jarkkojs** [rust-lang/rust#127868]
In my Fedora installation `/home/jarkko/.cargo/registry/src/index.crates.io-6f17d22bba15001f/vm-memory-0.14.1/src/lib.rs` ended up being a corrupted file.

It took me hours to realize, given that the output does not bring any clarity to it. rustc should really have ratelimit detection or something to detect corrupted file instead and inform about this to the user.

![Screenshot from 2024-07-17 17-04-06](https://github.com/user-attachments/assets/777798fb-3965-4e2a-9c59-c5002c1af210)

**Comment by Urgau** [comment]
This indeed seems quite confusing; and counter-productive to try to show this blob of errors.

Would you be able to provide us with a reproducible example of the issue? Or if you still have it the corrupted file.

@rustbot labels +T-compiler +A-diagnostics +D-verbose +D-confusing +S-needs-repro -needs-triage

**Comment by jarkkojs** [comment]
> This indeed seems quite confusing; and counter-productive to try to show this blob of errors.
> 
> Would you be able to provide us with a reproducible example of the issue? Or if you still have it the corrupted file.
> 
> @rustbot labels +T-compiler +A-diagnostics +D-verbose +D-confusing +S-needs-repro -needs-triage

So this a problem because I unfortunately deleted the trashed file before someone told me that I should report this as a bug. I regret that I did not store it :-/

BUT I recalled that it was probably for reason a long string of ANSI escape codes. So I did a test string:

```
\1\e[38;5;015m\2uu\1\e[0m\2\1\e[38;2;191;254;33m\2(ins)\1\e[38;5;015m\2uu\1\e[0m\2\1\e[38;2;191;254;33m\2(ins)\1\e[38;5;015m\2uu\1\e[0m\2\1\e[38;2;191;254;33m\2(ins)\1\e[38;5;015m\2uu\1\e[0m\2\1\e[38;2;191;254;33m\2(ins)\1\e[38;5;015m\2uu\1\e[0m\2\1\e[38;2;191;254;33m\2(ins)\1\e[38;5;015m\2uu\1\e[0m\2\1\e[38;2;191;254;33m\2(ins)\1\e[38;5;015m\2uu\1\e[0m\2\1\e[38;2;191;254;33m\2(ins)\1\e[38;5;015m\2uu\1\e[0m\2\1\e[38;2;191;254;33m\2(ins)\1\e[38;5;015m\2uu\1\e[0m\2\1\e[38;2;191;254;33m\2(ins)\1\e[38;5;015m\2uu\1\e[0m\2\1\e[38;2;191;254;33m\2(ins)\1\e[38;5;015m\2uu\1\e[0m\2\1\e[38;2;191;254;33m\2(ins)\1\e[38;5;015m\2uu\1\e[0m\2\1\e[38;2;191;254;33m\2(ins)\1\e[38;5;015m\2uu\1\e[0m\2\1\e[38;2;191;254;33m\2(ins)\1\e[38;5;015m\2uu\1\e[0m\2\1\e[38;2;191;254;33m\2(ins)\1\e[38;5;015m\2uu\1\e[0m\2\1\e[38;2;191;254;33m\2(ins)\1\e[38;5;015m\2uu\1\e[0m\2\1\e[38;2;191;254;33m\2(ins)\1\e[38;5;015m\2uu\1\e[0m\2\1\e[38;2;191;254;33m\2(ins)\1\e[38;5;015m\2uu\1\e[0m\2\1\e[38;2;191;254;33m\2(ins)\1\e[38;5;015m\2uu\1\e[0m\2\1\e[38;2;191;254;33m\2(ins)\1\e[38;5;015m\2uu\1\e[0m\2\1\e[38;2;191;254;33m\2(ins)\1\e[38;5;015m\2uu\1\e[0m\2\1\e[38;2;191;254;33m\2(ins)\1\e[38;5;015m\2uu\1\e[0m\2\1\e[38;2;191;254;33m\2(ins)\1\e[38;5;015m\2uu\1\e[0m\2\1\e[38;2;191;254;33m\2(ins)\1\e[38;5;015m\2uu\1\e[0m\2\1\e[38;2;191;254;33m\2(ins)\1\e[38;5;015m\2uu\1\e[0m\2\1\e[38;2;191;254;33m\2(ins)\1\e[38;5;015m\2uu\1\e[0m\2\1\e[38;2;191;254;33m\2(ins)\1\e[38;5;015m\2uu\1\e[0m\2\1\e[38;2;191;254;33m\2(ins)\1\e[38;5;015m\2uu\1\e[0m\2\1\e[38;2;191;254;33m\2(ins)
```
This creates somewhat similar output as I got, except probably the string was a bit longer in my case.

**Comment by jarkkojs** [comment]
OK doubling that starts to be almost perfect match:
```
\1\e[38;5;015m\2uu\1\e[0m\2\1\e[38;2;191;254;33m\2(ins)\1\e[38;5;015m\2uu\1\e[0m\2\1\e[38;2;191;254;33m\2(ins)\1\e[38;5;015m\2uu\1\e[0m\2\1\e[38;2;191;254;33m\2(ins)\1\e[38;5;015m\2uu\1\e[0m\2\1\e[38;2;191;254;33m\2(ins)\1\e[38;5;015m\2uu\1\e[0m\2\1\e[38;2;191;254;33m\2(ins)\1\e[38;5;015m\2uu\1\e[0m\2\1\e[38;2;191;254;33m\2(ins)\1\e[38;5;015m\2uu\1\e[0m\2\1\e[38;2;191;254;33m\2(ins)\1\e[38;5;015m\2uu\1\e[0m\2\1\e[38;2;191;254;33m\2(ins)\1\e[38;5;015m\2uu\1\e[0m\2\1\e[38;2;191;254;33m\2(ins)\1\e[38;5;015m\2uu\1\e[0m\2\1\e[38;2;191;254;33m\2(ins)\1\e[38;5;015m\2uu\1\e[0m\2\1\e[38;2;191;254;33m\2(ins)\1\e[38;5;015m\2uu\1\e[0m\2\1\e[38;2;191;254;33m\2(ins)\1\e[38;5;015m\2uu\1\e[0m\2\1\e[38;2;191;254;33m\2(ins)\1\e[38;5;015m\2uu\1\e[0m\2\1\e[38;2;191;254;33m\2(ins)\1\e[38;5;015m\2uu\1\e[0m\2\1\e[38;2;191;254;33m\2(ins)\1\e[38;5;015m\2uu\1\e[0m\2\1\e[38;2;191;254;33m\2(ins)\1\e[38;5;015m\2uu\1\e[0m\2\1\e[38;2;191;254;33m\2(ins)\1\e[38;5;015m\2uu\1\e[0m\2\1\e[38;2;191;254;33m\2(ins)\1\e[38;5;015m\2uu\1\e[0m\2\1\e[38;2;191;254;33m\2(ins)\1\e[38;5;015m\2uu\1\e[0m\2\1\e[38;2;191;254;33m\2(ins)\1\e[38;5;015m\2uu\1\e[0m\2\1\e[38;2;191;254;33m\2(ins)\1\e[38;5;015m\2uu\1\e[0m\2\1\e[38;2;191;254;33m\2(ins)\1\e[38;5;015m\2uu\1\e[0m\2\1\e[38;2;191;254;33m\2(ins)\1\e[38;5;015m\2uu\1\e[0m\2\1\e[38;2;191;254;33m\2(ins)\1\e[38;5;015m\2uu\1\e[0m\2\1\e[38;2;191;254;33m\2(ins)\1\e[38;5;015m\2uu\1\e[0m\2\1\e[38;2;191;254;33m\2(ins)\1\e[38;5;015m\2uu\1\e[0m\2\1\e[38;2;191;254;33m\2(ins)\1\e[38;5;015m\2uu\1\e[0m\2\1\e[38;2;191;254;33m\2(ins)\1\e[38;5;015m\2uu\1\e[0m\2\1\e[38;2;191;254;33m\2(ins)\1\e[38;5;015m\2uu\1\e[0m\2\1\e[38;2;191;254;33m\2(ins)\1\e[38;5;015m\2uu\1\e[0m\2\1\e[38;2;191;254;33m\2(ins)\1\e[38;5;015m\2uu\1\e[0m\2\1\e[38;2;191;254;33m\2(ins)\1\e[38;5;015m\2uu\1\e[0m\2\1\e[38;2;191;254;33m\2(ins)\1\e[38;5;015m\2uu\1\e[0m\2\1\e[38;2;191;254;33m\2(ins)\1\e[38;5;015m\2uu\1\e[0m\2\1\e[38;2;191;254;33m\2(ins)\1\e[38;5;015m\2uu\1\e[0m\2\1\e[38;2;191;254;33m\2(ins)\1\e[38;5;015m\2uu\1\e[0m\2\1\e[38;2;191;254;33m\2(ins)\1\e[38;5;015m\2uu\1\e[0m\2\1\e[38;2;191;254;33m\2(ins)\1\e[38;5;015m\2uu\1\e[0m\2\1\e[38;2;191;254;33m\2(ins)\1\e[38;5;015m\2uu\1\e[0m\2\1\e[38;2;191;254;33m\2(ins)\1\e[38;5;015m\2uu\1\e[0m\2\1\e[38;2;191;254;33m\2(ins)\1\e[38;5;015m\2uu\1\e[0m\2\1\e[38;2;191;254;33m\2(ins)\1\e[38;5;015m\2uu\1\e[0m\2\1\e[38;2;191;254;33m\2(ins)\1\e[38;5;015m\2uu\1\e[0m\2\1\e[38;2;191;254;33m\2(ins)\1\e[38;5;015m\2uu\1\e[0m\2\1\e[38;2;191;254;33m\2(ins)\1\e[38;5;015m\2uu\1\e[0m\2\1\e[38;2;191;254;33m\2(ins)\1\e[38;5;015m\2uu\1\e[0m\2\1\e[38;2;191;254;33m\2(ins)\1\e[38;5;015m\2uu\1\e[0m\2\1\e[38;2;191;254;33m\2(ins)\1\e[38;5;015m\2uu\1\e[0m\2\1\e[38;2;191;254;33m\2(ins)\1\e[38;5;015m\2uu\1\e[0m\2\1\e[38;2;191;254;33m\2(ins)\1\e[38;5;015m\2uu\1\e[0m\2\1\e[38;2;191;254;33m\2(ins)\1\e[38;5;015m\2uu\1\e[0m\2\1\e[38;2;191;254;33m\2(ins)\1\e[38;5;015m\2uu\1\e[0m\2\1\e[38;2;191;254;33m\2(ins)\1\e[38;5;015m\2uu\1\e[0m\2\1\e[38;2;191;254;33m\2(ins)\1\e[38;5;015m\2uu\1\e[0m\2\1\e[38;2;191;254;33m\2(ins)\1\e[38;5;015m\2uu\1\e[0m\2\1\e[38;2;191;254;33m\2(ins)
```

**Comment by jarkkojs** [comment]
![Screenshot from 2024-07-17 22-57-28](https://github.com/user-attachments/assets/3b97ab42-3f4e-4899-a42a-25eafc9c018f)

... I'm considering T-shirt design ;-) Made by me and rustc.

**Comment by jarkkojs** [comment]
Somehow rustc should be able to detect corrupted file in cargo's source repository (or whatever it is called). It was pretty bizarre sight when I encountered this by accident.

**Comment by Urgau** [comment]
Thanks for the repro.
@rustbot label -S-needs-repro

**Comment by jarkkojs** [comment]
Yeah I mean it is really a huge corner case but it can happen in the case of data corruption (e.g. power outage in middle of write operation). I'd detect frequency of errors and maybe a threshold parameter when it is interpreted as corruption with a sane default value.

**Comment by chenyukang** [comment]
yeah, I believe this error comes out from the phase of lexing, which rustc are trying to find all the mismatched delimiters, maybe a threshold is ok for resolve it.
